x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java

```diff
@@ -74,16 +83,53 @@ protected QueryBuilder parseQuery(XContentParser parser) throws IOException {
         return parseInnerQueryBuilder;
     }
 
-    private List<String> randomSearchAfter() {
-        if (randomBoolean()) {
-            return Collections.emptyList();
-        } else {
-            int size = randomIntBetween(1, 50);
-            List<String> arr = new ArrayList<>(size);
-            for (int i = 0; i < size; i++) {
-                arr.add(randomAlphaOfLength(randomIntBetween(1, 15)));
+    private SearchAfterBuilder randomJsonSearchFromBuilder() throws IOException {
+        int numSearchAfter = randomIntBetween(1, 10);
+        XContentBuilder jsonBuilder = XContentFactory.jsonBuilder();
+        jsonBuilder.startObject();
+        jsonBuilder.startArray("search_after");
+        for (int i = 0; i < numSearchAfter; i++) {
+            int branch = randomInt(9);
```

Comment:
I think there is `LuceneTests.randomSortValue()` that generates random sort values. I wonder if we could use it here.

Reply:
If it is not possible, I think there is a [better pattern for this](https://github.com/elastic/elasticsearch/blob/76de93c25808b51b27f5d3cedaee159cf3d74fea/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractorTests.java#L588) base on `randomFrom(...)` and `Supplier<...>`

Reply:
updated to a better pattern